src/slurm_plugin/clustermgtd.py

```diff
@@ -1126,37 +901,127 @@ def _handle_protected_mode_process(self, slurm_nodes, partitions, bootstrap_fail
         # Partition does not have job running
         log.info(
             "Partitions bootstrap failure count: %s, cluster will be set into protected mode if "
-            "protected failure count reach threshold",
-            self._partitions_protected_failure_count_map,
+            "protected failure count reach threshold.", self._partitions_protected_failure_count_map
         )
+        inactive_partitions = get_inactive_partition_names(partition_names_map)
         partitions_above_threshold = [
             part
             for part, failures in self._partitions_protected_failure_count_map.items()
-            if part not in self._inactive_partitions and failures >= self._config.protected_failure_count
+            if part not in inactive_partitions and failures >= self._config.protected_failure_count
         ]
         if partitions_above_threshold:
             log.info("Bootstrap failure partitions: %s", print_with_count(partitions_above_threshold))
-            nodename_to_slurm_nodes_map = {node.name: node for node in slurm_nodes}
             partitions_to_disable = ClusterManager._get_partitions_to_disable(
-                nodename_to_slurm_nodes_map, partitions_above_threshold, partitions
+                partitions_above_threshold, partition_names_map
             )
             self._enter_protected_mode(partitions_to_disable)
 
     @staticmethod
-    def _filter_partitions_with_active_jobs(nodename_to_slurm_nodes_map, partitions):
+    def _filter_partitions_with_active_jobs(partition_names_map):
         """Retrieve partitions currently have active jobs running."""
         partitions_have_active_jobs = set()
-        for partition_name, partition in partitions.items():
-            partiton_nodenames = get_partition_node_names(partition.nodes)
-            for nodename in partiton_nodenames:
-                if (
-                    nodename_to_slurm_nodes_map.get(nodename)
-                    and nodename_to_slurm_nodes_map.get(nodename).is_running_job()
-                ):
+        for partition_name, partition in partition_names_map.items():
+            for node in partition.slurm_nodes:
+                if node.is_running_job():
                     partitions_have_active_jobs.add(partition_name)
                     break
         return partitions_have_active_jobs
 
+    @staticmethod
+    def _map_instances_and_nodes(active_nodes, inactive_nodes, cluster_instances):
+        """Initialize the instance behind slurm nodes."""
+        # Initialize mapping data structures
+        active_ip_to_slurm_node_map = {node.nodeaddr: node for node in active_nodes}
+        log.debug("ip_to_slurm_node_map %s", active_ip_to_slurm_node_map)
+        inactive_ip_to_slurm_node_map = {node.nodeaddr: node for node in inactive_nodes}
```

Comment:
Can we do these two together? Like `{node.nodeaddr: node for node in active_nodes + inactive_nodes}`?

Reply:
Ok, in that case we don't actually need `inactive_ip_to_slurm_node_map`? We might need a set of `inactive_ips` but we don't need the `inactive_ip_to_slurm_node_map`?